qiskit/circuit/classicalfunction/boolean_expression.py

```diff
@@ -0,0 +1,122 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A quantum oracle constructed from a logical expression or a string in the DIMACS format."""
+
+from os.path import basename, isfile
+
+from qiskit.exceptions import MissingOptionalLibraryError
+from .classical_element import ClassicalElement
+from .utils import HAS_TWEEDLEDUM
+
+
+class BooleanExpression(ClassicalElement):
+    """The Boolean Expression gate."""
+
+    def __init__(self, expression: str, name: str = None) -> None:
```

Comment:
`name` should have the type hint `Optional[str] = None` to show that `None` is also a valid value

Reply:
done in 15f48136d